src/Reflection/ReflectionClass.php

```diff
@@ -664,21 +665,52 @@ public function getImmediateProperties(?int $filter = null): array
     {
         if ($this->cachedImmediateProperties === null) {
             $properties = [];
-            foreach ($this->node->stmts as $stmt) {
-                if (! ($stmt instanceof PropertyNode)) {
+            foreach ($this->node->getProperties() as $propertiesNode) {
+                foreach ($propertiesNode->props as $propertyPositionInNode => $propertyNode) {
+                    $property                         = ReflectionProperty::createFromNode(
+                        $this->reflector,
+                        $propertiesNode,
+                        $propertyPositionInNode,
+                        $this->declaringNamespace,
+                        $this,
+                        $this,
+                        false,
+                    );
+                    $properties[$property->getName()] = $property;
+                }
+            }
+
+            foreach ($this->node->getMethods() as $methodNode) {
```

Comment:
Is there a faster way to find the ctor, than looping through these? (fine also if not)

Reply:
I think this is the most optimized way. We don't need to create any reflection, just iterate existing AST.